pkg/apis/batch/v1/defaults_test.go

```diff
@@ -245,3 +245,69 @@ func roundTrip(t *testing.T, obj runtime.Object) runtime.Object {
 	}
 	return obj3
 }
+
+func TestSetDefaultCronJob(t *testing.T) {
+	tests := map[string]struct {
+		original *batchv1.CronJob
+		expected *batchv1.CronJob
+	}{
+		"empty batchv1.CronJob should default batchv1.ConcurrencyPolicy and Suspend": {
+			original: &batchv1.CronJob{},
+			expected: &batchv1.CronJob{
+				Spec: batchv1.CronJobSpec{
+					ConcurrencyPolicy:          batchv1.AllowConcurrent,
+					Suspend:                    newBool(false),
+					SuccessfulJobsHistoryLimit: utilpointer.Int32Ptr(3),
+					FailedJobsHistoryLimit:     utilpointer.Int32Ptr(1),
+				},
+			},
+		},
+		"set fields should not be defaulted": {
+			original: &batchv1.CronJob{
+				Spec: batchv1.CronJobSpec{
+					ConcurrencyPolicy:          batchv1.ForbidConcurrent,
+					Suspend:                    newBool(true),
+					SuccessfulJobsHistoryLimit: utilpointer.Int32Ptr(5),
+					FailedJobsHistoryLimit:     utilpointer.Int32Ptr(5),
+				},
+			},
+			expected: &batchv1.CronJob{
+				Spec: batchv1.CronJobSpec{
+					ConcurrencyPolicy:          batchv1.ForbidConcurrent,
+					Suspend:                    newBool(true),
+					SuccessfulJobsHistoryLimit: utilpointer.Int32Ptr(5),
+					FailedJobsHistoryLimit:     utilpointer.Int32Ptr(5),
+				},
+			},
+		},
+	}
+
+	for name, test := range tests {
+		original := test.original
+		expected := test.expected
+		obj2 := roundTrip(t, runtime.Object(original))
+		actual, ok := obj2.(*batchv1.CronJob)
+		if !ok {
+			t.Errorf("%s: unexpected object: %v", name, actual)
+			t.FailNow()
+		}
+		if actual.Spec.ConcurrencyPolicy != expected.Spec.ConcurrencyPolicy {
```

Comment:
nit: replace all these checks with a single `cmp.Diff`?

Reply:
I don't see any value in doing this.

Reply:
I think in those cases I think it's better being explicit and catch all mismatches during tests, not gracefully handle nil which should not happen.

Reply:
`Diff` gives you a report of all the differences. When I say gracefully handle nil, I mean that it will report in one field is nil and not the other, but it won't crash. The existing code would just crash.

Reply:
Yup, I'm aware of that :smile:

Reply:
Then why is this not desired? We tend to advocate for `cmp.Diff` in scheduler tests as the code becomes easier to maintain. Are we missing something?